lib/irslackd.js

```diff
@@ -88,24 +89,68 @@ class Irslackd {
   }
   async onIrcAway(ircUser, msg) {
       // if args is empty, unset away status
-      let status;
+      let status; let code; let reply;
       if (msg.args.length == 0) {
           status = '';
           this.setSlackPresence(ircUser, true);
+          code = '305';
+          reply = ':You are no longer marked as being away';
       } else {
           status = msg.args[0];
           // slack statuses are capped at 100 characters
           status = status.substring(0, 100);
           this.setSlackPresence(ircUser, false);
+          code = '306';
+          reply = ':You have been marked as being away'
       }
       let profile;
       try {
           profile = await ircUser.slackWeb.apiCall('users.profile.set', { profile: { status_text: status , status_emoji: '' } });
           if (!profile.ok) throw profile;
+          this.ircd.write(ircUser.socket, 'irslackd', code, [ '=', reply ]);
       } catch(e) {
           this.logError(ircUser, 'Failed users.profile.set: ' + util.inspect(e));
       }
   }
+  async onIrcInvite(ircUser, msg) {
+      let code;
+      let reply;
+
+      if (msg.args.length < 2) {
+          code = '461'; // ERR_NEEDMOREPARAMS
+          reply = 'INVITE :Not enough parameters';
+      }
+      else {
+          let ircNick = msg.args[0];
+          let ircChan = msg.args[1];
+
+          if (!ircUser.ircToSlack.has(ircChan) || !ircUser.inChannel.get(ircChan)) {
+              code = '442'; // ERR_NOTONCHANNEL
+              reply = ircChan + ':You\'re not on that channel';
+          }
+          else if (!ircUser.ircToSlack.has(ircNick)) {
+              code = '401'; // ERR_NOSUCHNICK
+              reply = ircNick + ' :No such nick/channel';
+          }
+          else {
+              code = '341'; // RPL_INVITING
+              reply = ircChan +' '+ ircNick;
+
+              let slackId = ircUser.ircToSlack.get(ircNick);
+              let slackChan = ircUser.ircToSlack.get(ircChan);
+
+              let invite;
+              try {
+                  invite = await ircUser.slackWeb.apiCall('channels.invite', { channel: slackChan, user: slackId });
+                  if (!invite.ok) throw invite;
+              } catch(e) {
+                  this.logError(ircUser, 'Failed channels.invite: ' + util.inspect(e));
+                  return;
```

Comment:
I couldn't find a fitting numeric response for when there is a failed API call. Open to suggestions on what to reply with in this case